pkg/spec/config_linux.go

```diff
@@ -112,36 +113,41 @@ func (c *CreateConfig) createBlockIO() (*spec.LinuxBlockIO, error) {
 			lwds = append(lwds, lwd)
 		}
 		bio.WeightDevice = lwds
+		ret = bio
 	}
 	if len(c.Resources.DeviceReadBps) > 0 {
 		readBps, err := makeThrottleArray(c.Resources.DeviceReadBps, bps)
 		if err != nil {
 			return bio, err
 		}
 		bio.ThrottleReadBpsDevice = readBps
+		ret = bio
 	}
 	if len(c.Resources.DeviceWriteBps) > 0 {
 		writeBpds, err := makeThrottleArray(c.Resources.DeviceWriteBps, bps)
 		if err != nil {
 			return bio, err
 		}
 		bio.ThrottleWriteBpsDevice = writeBpds
+		ret = bio
 	}
 	if len(c.Resources.DeviceReadIOps) > 0 {
 		readIOps, err := makeThrottleArray(c.Resources.DeviceReadIOps, iops)
 		if err != nil {
 			return bio, err
 		}
 		bio.ThrottleReadIOPSDevice = readIOps
+		ret = bio
 	}
 	if len(c.Resources.DeviceWriteIOps) > 0 {
 		writeIOps, err := makeThrottleArray(c.Resources.DeviceWriteIOps, iops)
 		if err != nil {
 			return bio, err
 		}
 		bio.ThrottleWriteIOPSDevice = writeIOps
+		ret = bio
 	}
-	return bio, nil
+	return ret, nil
```

Comment:
Maybe I'm lacking tea, but I'm not sure how the ret = bio dance helps here?

Reply:
I agree, it looks like a no-op patch overall.

Reply:
ret is initialized to `nil`.  It is set to `&bio` only if we are really setting something in the function